src/main/java/frc/robot/commands/AlignToPose.java

```diff
@@ -0,0 +1,109 @@
+import com.ctre.phoenix6.swerve.SwerveRequest;
+import dev.doglog.DogLog;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.generated.TunerConstants;
+import frc.robot.subsystems.CommandSwerveDrivetrain;
+import java.util.function.Supplier;
+
+public class AlignToPose extends Command {
+  Pose2d currPose;
+  double angularVelocity;
+  DriveCommand driveCommand;
+  Supplier<Pose2d> targetPose;
+  private PIDController PIDX;
+  private PIDController PIDY;
+  private PIDController PIDRotation;
+  private CommandSwerveDrivetrain drivetrain;
+  private double MaxSpeed =
+      TunerConstants.kSpeedAt12Volts.in(MetersPerSecond); // kSpeedAt12VoltsMps desired top speed
+  private double MaxAngularRate =
+      3.5 * Math.PI; // 3/4 of a rotation per second max angular velocity
+  private final SwerveRequest.FieldCentric drive =
+      new SwerveRequest.FieldCentric()
+          .withDeadband(MaxSpeed * 0.1)
+          .withRotationalDeadband(MaxAngularRate * 0.1) // Add a 10% deadband
+          .withDriveRequestType(DriveRequestType.OpenLoopVoltage); // I want field-centric
+
+  public AlignToPose(Supplier<Pose2d> Pose, CommandSwerveDrivetrain drivetrain) {
+    addRequirements(drivetrain);
+    targetPose = Pose;
+    PIDX = new PIDController(2.7, 0, 0); // same for now tune later
+    PIDX.setTolerance(0.1);
+    PIDY = new PIDController(2.7, 0, 0);
+    PIDY.setTolerance(0.1);
+    PIDRotation = new PIDController(0.002, 0, 0);
+    PIDRotation.setTolerance(0.1);
+    PIDRotation.enableContinuousInput(Radians.fromBaseUnits(-180), Radians.fromBaseUnits(180));
+    this.drivetrain = drivetrain;
+  }
+
+  public void goToPoseWithPID(Pose2d targetPose) {
+    PIDX.setSetpoint(targetPose.getX());
+    PIDY.setSetpoint(targetPose.getY());
+    PIDRotation.setSetpoint(targetPose.getRotation().getRadians());
+  }
+
+  public boolean isAtTargetPose() {
+    if (PIDX.atSetpoint() && PIDY.atSetpoint() && PIDRotation.atSetpoint()) {
+      return true;
+    }
+    return false;
+  }
+
+  @Override
+  public void initialize() {
+    goToPoseWithPID(targetPose.get());
+  }
+
+  @Override
+  public void execute() {
+    double xVelocity = 0;
+    double yVelocity = 0;
+    double angularVelocity = 0;
+    currPose = drivetrain.getState().Pose;
+    double currX = currPose.getX();
+    double currY = currPose.getY();
+    Double currRotation = currPose.getRotation().getRadians();
+    double PIDXOutput = PIDX.calculate(currX);
+    xVelocity -= PIDXOutput;
+    DogLog.log("PIDXOutput", PIDXOutput);
+    double PIDYOutput = PIDY.calculate(currY);
+    yVelocity -= PIDYOutput;
+    DogLog.log("PIDYoutput", PIDYOutput);
+    double PIDRotationOutput = PIDRotation.calculate(currRotation);
+    angularVelocity += PIDRotationOutput;
+    DogLog.log("PIDRotationoutput", PIDRotationOutput);
+    xVelocity = xVelocity * MaxSpeed;
```

Comment:
The scaling of the x/y/theta values here was useful for the DriveCommand mode, because it used the input from the joystick, which ranged from-1 to +1.   I'm not clear it makes sense in the self driving mode, since the output of the PIDs is in a different domain of numbers.    We should look at those numbers in logs to understand.

We may need to implement a speed scaling value for testing purposes too, so we can have more safety in testing situations.